app/controllers/api/v1/card_decks_controller.rb

```diff
@@ -2,21 +2,18 @@ class Api::V1::CardDecksController < ApplicationController
   
   # GET /api/v1/card_decks
   def index
-    card_decks = CardDeck.all
-    card_decks_json = ActiveModel::Serializer::CollectionSerializer.new(card_decks, each_serializer: CardSerializer)
+    authorize CardDeck
+    
+    card_decks = CardDeck.joins(:card, :deck).where(cards: { user: current_user }).where(decks: { user: current_user }).order(id: :desc)
```

Comment:
This is not correct, this query grabs all CardDecks where user is both "owner of Card" AND "owner of Deck".
It should be "owner of Card" OR "owner of Deck".